pydoclint/main.py

```diff
@@ -166,13 +166,15 @@ def main(  # noqa: C901
     if paths and src is not None:
         click.echo(
             main.get_usage(ctx)
-            + "\n\n'paths' and 'src' cannot be passed simultaneously."
+            + "\n\n'paths' and 'src' cannot be passed simultaneously.",
+            err=True,
```

Comment:
Hi @s-weigand , from your comment [here](https://github.com/jsh9/pydoclint/issues/20#issuecomment-1573718002), I understand why we are setting `err` to True.  But other future readers of this code may not realize it.

To avoid potential confusion in the future, may I suggest this:

```python
from sys import platform

# In Windows, we echo the messages as stderr in order to bypass
# a known issue with Windows + pre-commit
echoAsError: bool = platform == 'win32'

...

click.echo(..., err=echoAsError)
```

Reply:
Since linter output is considered an error IMHO we should follow the example of black and output all to `stderr`, that way users also won't accidentally pipe linter output into files.
Even so, the reason for this change is an issue on Windows I think OS dependent behavior is a bad UX and makes using tools and guiding contributors for projects using `pydoclint` much harder.

Reply:
Even though I do not completely agree that "inter output is considered an error", I think what you said about "OS dependent behavior is a bad UX" made a ton of sense.

That said, I think we still need to explain why we set `err=True` in all places.  Therefore, I'm going to make a code change like this:

```python
# Due to a potential bug in Windows + pre-commit, non-ASCII
# characters cannot be rendered correctly as stdout in the terminal.
# Therefore, we set all CLI output as stderr.
echoAsError: True

...

click.echo(..., err=echoAsError)
```